messages.go

```diff
@@ -121,12 +153,58 @@ type Data struct {
 	Content    []byte            `json:"content"`
 }
 
+func (d *Data) String() string {
+	result := fmt.Sprintf("{Type:%s MessageID:%s", d.Type, d.MessageID)
+	if len(d.ResponseTo) > 0 {
+		result += fmt.Sprintf(" ResponseTo:%s", d.ResponseTo)
+	}
+	result += fmt.Sprintf(
+		" Version:%d Sent:%s Directive:%s Metadata:%v",
+		d.Version, d.Sent.Format(time.RFC3339), d.Directive, d.Metadata,
+	)
+
+	if isASCII(d.Content) {
+		result += fmt.Sprintf(" Content:%s", string(d.Content))
+	} else {
+		result += fmt.Sprintf(" Content:%v", d.Content)
+	}
+
+	result += "}"
+	return result
+}
+
 // Response messages are published by the server as a response to a data
-// message. This is most often used as a receipt to indicate the receiption of a
+// message. This is most often used as a receipt to indicate the reception of a
 // message by a synchronous request/response transport (such as the HTTP polling
 // transport).
 type Response struct {
 	Code     int               `json:"code"`
 	Metadata map[string]string `json:"metadata"`
 	Data     []byte            `json:"data"`
 }
+
+func (r *Response) String() string {
+	result := fmt.Sprintf("{Code:%d Metadata:%v", r.Code, r.Metadata)
+
+	if isASCII(r.Data) {
+		result += fmt.Sprintf(" Data:%s", string(r.Data))
+	} else {
+		result += fmt.Sprintf(" Data:%v", r.Data)
+	}
+
+	result += "}"
+	return result
+}
+
+// isASCII detects whether a byte slice is an ASCII text or not.
+func isASCII(data []byte) bool {
```

Comment:
I'm not convinced that we want to use this heuristics, because you have to go through the whole message. 

Maybe we could add something like `Content-type` HTTP header to message. Then decision how to print content could be easier and more straightforward.

Reply:
Perhaps something like [QuoteToASCII](https://pkg.go.dev/strconv@go1.21.4#QuoteToASCII) is better here?